jobs/build-charms/charms.py

```diff
@@ -335,21 +371,26 @@ def setup(self):
         ):
             click.echo(line)
 
-    def proof_build(self):
+    def build(self):
         """Perform charm build against charm/bundle"""
         if "override-build" in self.opts:
             click.echo("Override build found, running in place of charm build.")
             ret = script(
                 self.opts["override-build"], cwd=self.src_path, charm=self.name
             )
+        elif self.legacy_charm:
+            ret = cmd_ok(
+                "charm build -r --force -i https://localhost",
+                cwd=self.src_path,
+            )
         else:
             ret = cmd_ok(
-                "charm build -r --force -i https://localhost", cwd=self.src_path
+                f"{self.build.home_dir}/.local/bin/charmcraft build -f {self.src_path}",
```

Comment:
This hard coded path seems odd. Is there not a path to charmcraft available somewhere?

Reply:
CI has se unique obstacles to deal with bc the home directory is not what snap can use. So we use pypi local installs which guarantee the path. I think this is fine